src/cgroups/v2/freezer.rs

```diff
@@ -0,0 +1,206 @@
+use anyhow::{Result, *};
+use std::{
+    fs::OpenOptions,
+    io::{BufRead, BufReader, Read, Seek, SeekFrom, Write},
+    path::Path,
+    str, thread,
+    time::Duration,
+};
+
+use oci_spec::{FreezerState, LinuxResources};
+
+use super::controller::Controller;
+
+const CGROUP_FREEZE: &str = "cgroup.freeze";
+const CGROUP_EVENTS: &str = "cgroup.events";
+
+pub struct Freezer {}
+
+impl Controller for Freezer {
+    fn apply(linux_resources: &LinuxResources, cgroup_path: &Path) -> Result<()> {
+        if let Some(freezer_state) = linux_resources.freezer {
+            Self::apply(freezer_state, cgroup_path)?;
+        }
+
+        Ok(())
+    }
+}
+
+impl Freezer {
+    fn apply(freezer_state: FreezerState, path: &Path) -> Result<()> {
+        let state_str;
+        match freezer_state {
+            FreezerState::Undefined => {
+                return Ok(());
+            }
+            FreezerState::Frozen => {
+                state_str = "1";
+            }
+            FreezerState::Thawed => {
+                state_str = "0";
+            }
+        }
+
+        let r = OpenOptions::new()
+            .create(false)
+            .write(true)
+            .open(path.join(CGROUP_FREEZE));
+        if r.is_err() {
+            // ignore error if it is not a freeze op.
+            if let FreezerState::Frozen = freezer_state {
+                bail!("freezer not support {}", r.unwrap_err());
+            }
+            return Ok(());
+        }
+
+        r.unwrap().write_all(state_str.as_bytes())?;
```

Comment:
How about using the match syntax?
```suggestion
        match OpenOptions::new()
            .create(false)
            .write(true)
            .open(path.join(CGROUP_FREEZE))
        {
            Err(err) => {
                // ignore error if it is not a freeze op.
                if let FreezerState::Frozen = freezer_state {
                    bail!("freezer not support {}", err);
                }
                return Ok(());
            }
            Ok(mut file) => file.write_all(state_str.as_bytes())?,
        }
```